### PR TITLE
add a class to provide some common entity permissions

### PR DESCRIPTION
**Pull request by dawehner**

### CODE REVIEW

**Comment by bojanz** [comment]
Good start.

We can detect if the entity type has a revision key defined, and only provide the revision permissions in that case.

The "access" permissions are weird, can we name them "view" to make it more consistent?
And do we need both a "bypass access" and an "admin" permission?

File: src/Access/EntityBasePermissions.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      $container->get('entity_type.manager'),
+      $container->get('entity_type.bundle.info')
+    );
+  }
+
+  /**
+   * Gets an array of entity type permissions.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityTypeInterface $entity_type
+   *   The custom entity definition.
```
Comment (fago) [inline src/Access/EntityBasePermissions.php]: Why custom?

File: src/Access/EntityBasePermissions.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        "administer $entity_type_id" => [
+          'title' => $this->t('Administer @entity_label', $replacements),
+          'restrict access' => TRUE,
+        ],
+        "access $entity_type_id overview" => [
+          'title' => $this->t('Access the @entity_label overview page', $replacements),
+          'description' => $this->t('Get an overview of all @entity_label.', $replacements),
+        ],
+        "access $entity_type_id" => [
+          'title' => $this->t('View published @entity_label', $replacements),
```
Comment (fago) [inline src/Access/EntityBasePermissions.php]: publishing is not necessarily a concept/wording used, sometimes it's just active/inactive. Would it make sense to use a more generic wording here?

File: src/Access/EntityBasePermissions.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+   *
+   * @param \Drupal\Core\Entity\ContentEntityTypeInterface $entity_type
+   *   The custom entity definition.
+   *
+   * @return array
+   *   The entity type permissions.
+   *
+   * @see \Drupal\user\PermissionHandlerInterface::getPermissions()
+   */
+  public function entityPermissions(ContentEntityTypeInterface $entity_type) {
```
Comment (fago) [inline src/Access/EntityBasePermissions.php]: get ? Also, is this really bound to content entities? If not, let's not unnecessarily type-hint on it?

File: src/Access/EntityBasePermissions.php [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+      "create $bundle_id $base_entity_type_id" => [
+        'title' => $this->t('%bundle_id: Create new %entity_label', $type_params),
+      ],
+      "edit own $bundle_id $base_entity_type_id" => [
+        'title' => $this->t('%bundle_id: Edit own %entity_label', $type_params),
+      ],
+      "edit any $bundle_id $base_entity_type_id" => [
+        'title' => $this->t('%bundle_id: Edit any %entity_label', $type_params),
+      ],
+      "delete own $bundle_id $base_entity_type_id" => [
```
Comment (fago) [inline src/Access/EntityBasePermissions.php]: we need to check whether the entity implements the owner interface here

**Comment by fago** [comment]
Usually, admin just means you can access administrative pages and bypass give you full access control, thus having two permissions make sense imo.

> The "access" permissions are weird, can we name them "view" to make it more consistent?
> We have the same problem with publishing vs active/inactive - permission names come with some assumptions on how the entity is used. We could just go with the common view/publish wording/use for now and work on making this more flexible as needed in follow-ups?

**Comment by dawehner** [comment]
Will work on the fixes now

**Comment by bojanz** [comment]
Is this ready for review again?

**Comment by dawehner** [comment]
As written in the other issue, this could be considered to be blocked on non existing access control handler code

**Comment by lemonlab** [comment]
Hello,

This would be very usefull for Drupal Commerce (product add permission for a marketplace).
See : https://www.drupal.org/node/2733253

What's left to do to provide this feature ? I'm ready to help if needed, but i don't know were to start.

**Comment by smmccabe** [comment]
There appears to be access handler code now so I don't think that's blocking anymore? like Francois i am willing to help out but I'm not sure what still needs to be done on this issue now.

**Comment by bojanz** [comment]
Outdated, see work in https://www.drupal.org/node/2801031

**Pull request closed without merging**